packages/terra-search-field/src/SearchField.jsx

```diff
@@ -162,8 +167,19 @@ class SearchField extends React.Component {
   }
 
   updateSearchText(searchText) {
-    if (searchText !== undefined && searchText !== this.state.searchText) {
-      this.setState({ searchText });
+    if (searchText !== undefined && searchText !== this.searchText) {
+      this.searchText = searchText;
+      // Forcing update for clearButton rerender.
+      this.forceUpdate();
+    }
+  }
+
+  handleInput(event) {
+    const textValue = event.target.value;
+    this.updateSearchText(textValue);
+
+    if (this.props.onInput) {
+      this.props.onInput(event, textValue);
```

Comment:
Sending back two parameters seems a little strange. We'll want to be as consistent as we can with these callback parameters. If we do decide to add this to the input field we'll have to add additional function logic to return the two parameters as opposed to just passing the function straight through to the input like we do the onChange.

https://github.com/cerner/terra-core/blob/7f1b23a3ff00608c4dcf47ceae24a47d2b94db4d/packages/terra-form-input/src/Input.jsx#L161-L163

Personally, I would keep this native. It's passive to add it in the future, but non-passive to take it away.

```jsx
    if (this.props.onInput) {
      this.props.onInput(event); // Only returning the event.
    }
```

Reply:
Should I do the same thing to our onChange handling in search-field as well? Because I was keeping it consistent with how the 'handleTextChange' function handles it's onChange prop https://github.com/cerner/terra-core/pull/2800/files#diff-105fddc0e7a42eb008cb5c09faad5222R161

Reply:
No, we'll want to stay passive with the onChange callback. We can grab a few more opinions on this before making any changes. 

My vote is `onInput(event)`.  We can passively add the searchText into the API in the future if it's requested.

Reply:
As per offline discussion, I have switched to only returning the even on the onInput prop call https://github.com/cerner/terra-core/pull/2800/commits/d02ec4b2a3e498a93dca1c78cf449791f00389d2